### PR TITLE
ssh_keys: support to create ssh keys for other users on client host

### PR DESCRIPTION
**Pull request by iccaszhulili**
Now it only support to create ssh key for current user.

Signed-off-by: Lily Zhu <lizhu@redhat.com>

### CODE REVIEW

File: virttest/virsh.py [hunk 1]
```diff
@@ -4857,3 +4857,37 @@ def guestinfo(name, options="", **dargs):
     :return: CmdResult object.
     """
     return command("guestinfo %s %s" % (name, options), **dargs)
+
+
+def get_user_sshkeys(name, user, **dargs):
```
Comment (chunfuwen) [inline virttest/virsh.py:4862]: Those code are already covered in https://github.com/avocado-framework/avocado-vt/pull/3058.

Comment (iccaszhulili) [reply virttest/virsh.py]: @updated

File: virttest/ssh_key.py [hunk 2]
```diff
@@ -10,17 +10,29 @@
...
 
 
-def get_public_key():
+def get_public_key(client_user=None):
     """
     Return a valid string ssh public key for the user executing autoserv or
     autotest. If there's no DSA or RSA public key, create a RSA keypair with
     ssh-keygen and return it.
 
+    :param client_user: genenrate the ssh_key for which client
```
Comment (chunfuwen) [inline virttest/ssh_key.py:19]: usually we prefer one line doc string for one parameter. Do you intend to use multiple lines ?

File: virttest/ssh_key.py [hunk 3]
```diff
@@ -43,8 +55,8 @@ def get_public_key():
 
     else:
         logging.info('Neither RSA nor DSA keypair found, creating RSA ssh key pair')
-        process.system('ssh-keygen -t rsa -q -N "" -f %s' %
-                       rsa_private_key_path)
+        process.system("su - %s -c 'ssh-keygen -t rsa -q -N \"\" -f %s'" %
```
Comment (chunfuwen) [inline virttest/ssh_key.py:58]: Do we need judge whether client_user is None or not ? it sound likes the string will be alike 'su - None' if client_user is None

Comment (iccaszhulili) [reply virttest/ssh_key.py]: Yes, you are right, updated.

**Comment by iccaszhulili** [comment]
Added a test
# avocado run --vt-type libvirt virsh.hostname.remote_connect
JOB ID     : ec6f913f086a260b5c45f0fc0461fdb668be5677
JOB LOG    : /root/avocado/job-results/job-2021-05-27T03.38-ec6f913/job.log
 (1/1) type_specific.io-github-autotest-libvirt.virsh.hostname.remote_connect: PASS (22.64 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 23.94 s

**Review by chunfuwen — APPROVED** [review]
lgtm

**Review by dzhengfy — APPROVED** [review]
LGTM

**Pull request merged**